Clarify S3 docs

gregorydlogan opened the pull request:
This pull request removes a reference to a now-deleted workflow from the AWS S3 archiving docs.  I've replaced the link with example WOH definitions instead.

Fixes: #4704

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] ~include migration scripts and documentation, if appropriate~
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

gregorydlogan committed:
Removng references to now-deleted workflow.  Replaced with example WOHs

mtneug commented on docs/guides/admin/docs/modules/awss3archive.md:
````diff
@@ -46,13 +46,33 @@ The Opencast AWS S3 Archive service configuration can be found in the
+        <configuration key="target-storage">local-filesystem</configuration>
+      </configurations>
+    </operation>
+```
````
I'm unsure if this "restore paragraph" is a good advice without more context. "Moving" is always based on snapshots. The `target-version` is used to specify what snapshot to move. It not specified (as done here) all snapshots are moved. This may be what is desired, but not necessarily.

The issue is that new snapshots are *always* created locally. You, therefor, mention that `move-storage` could be part of all workflows. Instead, I suggest using the tiered storage feature that moves older snapshots automatically to S3, e.g., after two weeks. Especially during the time of the ingest, there could be quite a lot of new snapshots because of metadata/ACL changes or workflows. The problem is that shapshots always contain all selected data, resulting in many downloads from S3.

When you know that a certain workflow will create a new shapshot, I would also not use `move-storage` as this would restore existing snapshots. Instead I would run `ingest-download` at the start of the workflow, which will put S3 files into the workspace. The `snapshot` operation is then very quick, but I think some operations also assume that everything is available locally (at least, I remember having problems with `sanitize-adaptive` when I did not download everything beforehand).

gregorydlogan replied:
Do you by any chance have a workflow that's already set up to do that?  That makes a ton of sense.

mtneug replied:
We still publish a reference to our workflows here (https://github.com/opencast/community-workflows/tree/master/13.x/educast.nrw), but they are completely different from upstream workflows and assume certain changes. However, the `ingest-download` workflow is rather simple: https://zivgitlab.uni-muenster.de/educast-nrw/opencast/workflows/-/blob/educast/13/src/workflows/educast-partial-setup-workspaces.yaml. This is included in every main workflow that will eventually create a snapshot or otherwise needs files locally.

gregorydlogan committed:
Additional work on S3 documentation per #4721 review.

mtneug commented on docs/guides/admin/docs/modules/awss3archive.md:
```diff
@@ -46,20 +46,48 @@ The Opencast AWS S3 Archive service configuration can be found in the
+S3 archiving is done on a Snapshot level, that is a mediapackage ID + version.  Because of the way that the Asset
+Manager handles snapshots, all newly created snapshots are *always* local.  Creating a snapshot of a mediapackage with
+non local data will download *all* related snapshots for that mediapackage which can incur significant costs.  S3
+archiving is meant to be a cost reduction, and storage expansion tool, rather than then primary storage for your
```
```suggestion
archiving is meant to be a cost reduction, and storage expansion tool, rather than the primary storage for your
```

mtneug commented on docs/guides/admin/docs/modules/awss3archive.md:
```diff
@@ -46,20 +46,48 @@ The Opencast AWS S3 Archive service configuration can be found in the
+archiving is meant to be a cost reduction, and storage expansion tool, rather than then primary storage for your
+asset manager.  Therefore, most adopters do not want to immediately (ie, at the end of your default workflow) offload
+your recordings to S3!  Instead, we suggest using the TimedMediaArchiver
+(org.opencastproject.assetmanager.impl.TimedMediaArchiver.cfg) to offload your recordings after sufficient time that
```
Is this picked up by the JavaScript config linking bit? Does that need `etc/` as prefix?

gregorydlogan replied:
It's not.  But I've reworked that bit, since it didn't read nicely.

mtneug reviewed (changes requested):
I think this reads nicely now. Although, I'm not sure what the definition of "primary storage" is. Main working space, sure. Place where most files are stored, probably not with S3. But I think the message comes across.

gregorydlogan committed:
Minor rewrite:
- Clarifies where the TimedMediaArchiver's config file should live
- Rewrites avoiding 'primary storage' since if you're offloading almost
  everything to S3 an argument could be made that S3 is your primary
  storage...
